src/daft-scan/src/scan_task_iters.rs

```diff
@@ -111,3 +116,92 @@ impl Iterator for MergeByFileSize {
         }
     }
 }
+
+pub fn split_by_row_groups(scan_tasks: BoxScanTaskIter, max_tasks: usize) -> BoxScanTaskIter {
+    let mut scan_tasks = itertools::peek_nth(scan_tasks);
+
+    // only split if we have a small amount of files
+    if scan_tasks.peek_nth(max_tasks).is_some() {
+        return Box::new(scan_tasks);
+    }
+
+    Box::new(
+        scan_tasks
+            .map(|t| -> DaftResult<BoxScanTaskIter> {
+                let t = t?;
+
+                // only split parquet tasks if they:
+                // - have one source
+                // - use native storage config
+                // - have no specified chunk spec or number of rows
+                match (
+                    t.file_format_config.as_ref(),
+                    t.storage_config.as_ref(),
+                    &t.sources[..],
+                    t.sources.get(0).map(DataFileSource::get_chunk_spec),
+                    t.pushdowns.limit,
+                ) {
+                    (
+                        FileFormatConfig::Parquet(_),
+                        StorageConfig::Native(_),
+                        [source],
+                        Some(None),
+                        None,
+                    ) => {
+                        let (io_runtime, io_client) =
+                            t.storage_config.get_io_client_and_runtime()?;
+
+                        let path = source.get_path();
+
+                        let io_stats =
+                            IOStatsContext::new(format!("split_by_row_groups for {:#?}", path));
+
+                        let runtime_handle = io_runtime.handle();
+
+                        let file = runtime_handle.block_on(read_parquet_metadata(
+                            path,
+                            io_client,
+                            Some(io_stats),
+                        ))?;
+
+                        Ok(Box::new(
+                            file.row_groups
+                                .into_iter()
+                                .enumerate()
+                                .map(|(i, row_group)| {
```

Comment:
I dont think we would want to do 1 ScanTask per row group but instead group row groups together as long as the size is under a threshold.

Reply:
I believe this is actually already implemented as the next part in the `scan_tasks` pipeline in the planner

https://github.com/Eventual-Inc/Daft/blob/kevin/split-parquet-row-groups/src/daft-plan/src/planner.rs#L89